camel/models/samba_model.py

```diff
@@ -0,0 +1,199 @@
+        self.model_config_dict = model_config_dict
+        self.check_model_config()
+
+    @property
+    def token_counter(self) -> BaseTokenCounter:
+        r"""Initialize the token counter for the model backend.
+
+        Returns:
+            BaseTokenCounter: The token counter following the model's
+                tokenization style.
+        """
+        if not self._token_counter:
+            self._token_counter = OpenAITokenCounter(ModelType.GPT_3_5_TURBO)
+        return self._token_counter
+
+    def check_model_config(self):
+        r"""Check whether the model configuration contains any
+        unexpected arguments to SambaNova API.
+
+        Raises:
+            ValueError: If the model configuration dictionary contains any
+                unexpected arguments to SambaNova API.
+        """
+        for param in self.model_config_dict:
+            if param not in SAMBA_API_PARAMS:
+                raise ValueError(
+                    f"Unexpected argument `{param}` is "
+                    "input into SambaNova model backend."
+                )
+
+    @api_keys_required("SAMBA_API_KEY")
+    def run(self, messages: List[OpenAIMessage]):
+        import json
+
+        import requests
+
+        url = "https://mzulfm3ekjd5j3hv.snova.ai/v1/chat/completions"
+
+        headers = {
+            "Authorization": f"Basic {self._api_key}",
+            "Content-Type": "application/json",
+        }
+
+        data = [
+            {
+                "messages": messages,
+                "max_tokens": self.token_limit,
+                "stop": ["<|eot_id|>"],
+                "model": self.model_type.value,
+                "stream": True,
+            }
+        ]
+
+        response = requests.post(
+            url, headers=headers, data=json.dumps(data), stream=True
+        )
+
+        # Check for successful response
+        if response.status_code != 200:
+            return {
+                "error": "Request failed",
+                "status_code": response.status_code,
+                "response_text": response.text,
+            }
+
+        result = []
+        for line in response.iter_lines():
+            if line:
+                line_str = line.decode('utf-8').strip()
+                # Remove "data: " prefix if it exists
+                if line_str.startswith('data: '):
+                    line_str = line_str[6:]
+                # Handle special end-of-stream marker
+                if line_str == '[DONE]':
+                    break
+                try:
+                    # Decode each line as JSON
+                    json_data = json.loads(line_str)
+                    result.append(json_data)
+                except json.JSONDecodeError:
+                    # Handle lines that are not valid JSON
+                    result.append(
+                        {"error": "Failed to decode line", "line": line_str}
+                    )
+
+        return self._to_openai_response(result)
```

Comment:
Naming consistency with other models
```suggestion
        return self._to_openai_response(reponse)
```